src/jp2image.cpp

```diff
@@ -501,10 +501,10 @@ namespace Exiv2
                             subBox.length = getLong((byte*)&subBox.length, bigEndian);
                             subBox.type = getLong((byte*)&subBox.type, bigEndian);
 
-                            // subBox.length makes no sense if it is larger than the rest of the file || 0
-                            if (subBox.length == 0 || subBox.length > io_->size() - io_->tell()) {
+                            if (subBox.length < sizeof(box) || subBox.length > io_->size() - io_->tell()) {
```

Comment:
Sorry for taking so long and not taking a proper look previously, but there's another issue further down. If you check the branch starting at line 518, you'll see that we are actually reading a user controlled amount of data from `data.pData_` (or at least 3 bytes).

We definitely must fix that one too (as that's another segfault waiting to happen), however I don't want to block this PR if you feel this is out of scope.

Reply:
I would suggest to create a ticket about that issue and solve it properly in other branch. When I addressed this issue I just tried to solve the bug described #742 so that we can have that fix in 0.27.1 soon.

I would like to address bigger refactorings and better testing approaches only in master, where we have more freedom to make bigger changes :innocent: